script/BaseGovernorDeployer.sol

```diff
@@ -22,19 +22,22 @@ abstract contract BaseGovernorDeployer is BaseDeployer, SharedGovernorConstants
 
   function run(address _implementation) public virtual returns (L2ArbitrumGovernorV2 _governor) {
     vm.startBroadcast(deployerPrivateKey);
-    TransparentUpgradeableProxy _proxy = new TransparentUpgradeableProxy(_implementation, L2_PROXY_ADMIN, "");
-    _governor = L2ArbitrumGovernorV2(payable(address(_proxy)));
-    _governor.initialize(
-      NAME(),
-      INITIAL_VOTING_DELAY,
-      INITIAL_VOTING_PERIOD,
-      INITIAL_PROPOSAL_THRESHOLD,
-      IVotes(L2_ARB_TOKEN_ADDRESS),
-      TimelockControllerUpgradeable(TIMELOCK_ADDRESS()),
-      QUORUM_NUMERATOR(),
-      INITIAL_VOTE_EXTENSION,
-      L2_UPGRADE_EXECUTOR
+    bytes memory _initData = abi.encodeCall(
+      L2ArbitrumGovernorV2.initialize,
+      (
+        NAME(),
+        INITIAL_VOTING_DELAY,
+        INITIAL_VOTING_PERIOD,
+        INITIAL_PROPOSAL_THRESHOLD,
+        IVotes(L2_ARB_TOKEN_ADDRESS),
+        TimelockControllerUpgradeable(TIMELOCK_ADDRESS()),
+        QUORUM_NUMERATOR(),
+        INITIAL_VOTE_EXTENSION,
+        L2_UPGRADE_EXECUTOR
+      )
     );
+    TransparentUpgradeableProxy _proxy = new TransparentUpgradeableProxy(_implementation, L2_PROXY_ADMIN, _initData);
+    _governor = L2ArbitrumGovernorV2(payable(address(_proxy)));
     vm.stopBroadcast();
```

Comment:
One thing to keep in mind for the future is this https://github.com/OpenZeppelin/openzeppelin-foundry-upgrades

Reply:
oh cool!